src/NServiceBus.Core/IBusExtensions.cs

```diff
@@ -107,7 +107,7 @@ public static void Unsubscribe(this IBus bus, Type messageType)
             Guard.AgainstNull("bus", bus);
             Guard.AgainstNull("messageType", messageType);
 
-            bus.Unsubscribe(messageType, new UnsubscribeOptions());
+            bus.UnsubscribeAsync(messageType, new UnsubscribeOptions());
```

Comment:
We need to fix this. The synchronous version should be obsoleted and only the async version should be exposed